BridgeAppSDK/SBAProfileItem.swift

```diff
@@ -0,0 +1,400 @@
+            guard let val = value as? NSNumber else { return }
+            self.value = val
+            
+        case SBAProfileTypeIdentifier.bool:
+            guard let val = value as? Bool else { return }
+            self.value = val
+            
+        case SBAProfileTypeIdentifier.date:
+            guard let stringVal = value as? String,
+                    let dateVal = NSDate(iso8601String: stringVal)
+                else { return }
+            self.value = dateVal
+            
+        case SBAProfileTypeIdentifier.hkBiologicalSex:
+            guard let val = value as? HKBiologicalSex else { return }
+            self.value = val
+            
+        case SBAProfileTypeIdentifier.hkQuantity:
+            guard let val = value as? NSNumber else { return }
+            self.value = val
+            
+        default:
+            break
+        }
+    }
+    
+    func commonDemographicJsonValue() -> SBBJSONValue? {
+        guard let jsonVal = self.commonJsonValueGetter() else { return nil }
+        if self.itemType == .hkBiologicalSex {
+            return (self.value as? HKBiologicalSex)?.demographicDataValue
+        }
+        
+        return jsonVal
+    }
+    
+    func commonCheckTypeCompatible(newValue: Any?) -> Bool {
+        guard newValue != nil else { return true }
+        
+        switch self.itemType {
+        case SBAProfileTypeIdentifier.string:
+            return newValue as? String != nil
+            
+        case SBAProfileTypeIdentifier.number:
+            return newValue as? NSNumber != nil
+            
+        case SBAProfileTypeIdentifier.bool:
+            return newValue as? NSNumber != nil
+            
+        case SBAProfileTypeIdentifier.date:
+            return newValue as? NSDate  != nil
+            
+        case SBAProfileTypeIdentifier.hkBiologicalSex:
+            return newValue as? HKBiologicalSex != nil
+            
+        case SBAProfileTypeIdentifier.hkQuantity:
+            return newValue as? NSNumber != nil
+            
+        default:
+            return false
+        }
+    }
+}
+
+class SBAProfileItemBase: NSObject, SBAProfileItem {
+    /**
+     The value property is used to get and set the profile item's value in whatever internal data
+     storage is used by the implementing class.
+     */
+    open var value: Any?
+
+    fileprivate let sourceDict: [AnyHashable: Any]
+    
+    open var profileKey: String {
+        get {
+            return sourceDict["profileKey"]! as! String
+        }
+    }
+    
+    open var sourceKey: String {
+        get {
+            return sourceDict["sourceKey"] as! String? ?? self.profileKey
+        }
+    }
+    
+    open var demographicKey: String {
+        get {
+            return sourceDict["demographicKey"] as! String? ?? self.profileKey
+        }
+    }
+    
+    open var itemType: SBAProfileTypeIdentifier {
+        get {
+            return sourceDict["itemType"]! as! SBAProfileTypeIdentifier
```

Comment:
Your itemType is a String whereas SBAProfileTypeIdentifier is a struct. Shouldn't this be:

````
return SBAProfileTypeIdentifier(rawValue: sourceDict["itemType"] as? String ?? "Unknown")
````

Or something like that?  Maybe:

````
guard let rawValue = sourceDict["itemType"] as? String 
else {
    return .string
}
return SBAProfileTypeIdentifier(rawValue: rawValue)
````